# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CompositionTransaction.h
# Commit: 1466d7d61102
# Full Hash: 1466d7d6110279cf167ab8b1567be693df5d7c3c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-03 09:50:30
# Regressor Bug: 1618089
# File Overlap Count: 2
# Description:
#   Bug 1618089 - part 7: Make some related methods use `EditorDOMPointInText` for arguments or result r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D64338
# ==============================================================================

diff -r 079f4b05ead2 -r 1466d7d61102 editor/libeditor/CompositionTransaction.h
--- a/editor/libeditor/CompositionTransaction.h	Tue Mar 03 02:19:59 2020 +0000
+++ b/editor/libeditor/CompositionTransaction.h	Tue Mar 03 01:15:38 2020 +0000
@@ -6,7 +6,9 @@
 #ifndef CompositionTransaction_h
 #define CompositionTransaction_h
 
-#include "mozilla/EditTransactionBase.h"   // base class
+#include "mozilla/EditTransactionBase.h"  // base class
+
+#include "mozilla/EditorDOMPoint.h"        // EditorDOMPointInText
 #include "nsCycleCollectionParticipant.h"  // various macros
 #include "nsString.h"                      // mStringToInsert
 
@@ -36,8 +38,8 @@
 class CompositionTransaction final : public EditTransactionBase {
  protected:
   CompositionTransaction(EditorBase& aEditorBase,
-                         const nsAString& aStringToInsert, dom::Text& aTextNode,
-                         uint32_t aOffset);
+                         const nsAString& aStringToInsert,
+                         const EditorDOMPointInText& aPointToInsert);
 
  public:
   NS_DECLARE_STATIC_IID_ACCESSOR(NS_IMETEXTTXN_IID)
@@ -52,13 +54,11 @@
    *                            be different from actual composition string.
    *                            E.g., password editor can hide the character
    *                            with a different character.
-   * @param aTextNode           The text node which will have aStringToInsert.
-   * @param aOffset             The offset in aTextNode where aStringToInsert
-   *                            will be inserted.
+   * @param aPointToInsert      The insertion point.
    */
   static already_AddRefed<CompositionTransaction> Create(
       EditorBase& aEditorBase, const nsAString& aStringToInsert,
-      dom::Text& aTextNode, uint32_t aOffset);
+      const EditorDOMPointInText& aPointToInsert);
 
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(CompositionTransaction,
                                            EditTransactionBase)